File: Тренировка 1.0/03. Множества/B. Пересечение множеств/Пересечение множества.cpp
#include <unordered_set>
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int number;
    unordered_set<int> set;
    do
    {
        cin >> number;
        set.insert(number);
    } while (cin.get() != '\n');

    vector<int> intersect;
    do
    {
        cin >> number;
        if (set.count(number)) intersect.push_back(number);
    } while (cin.get() != '\n');

    sort(intersect.begin(), intersect.end());
    for (int i = 0, number = intersect.size(); i < number; i++) cout << intersect[i] << ' ';
}

File: Тренировка 1.0/03. Множества/B. Пересечение множеств/Пересечение множества.cs
using System;
using System.Collections.Generic;

string[] array1 = Console.ReadLine().Split(' ');
string[] array2 = Console.ReadLine().Split(' ');

HashSet<int> set = new HashSet<int>();
for (int i = 0; i < array1.Length; i++) set.Add(int.Parse(array1[i]));

List<int> intersect = new List<int>();
for(int i = 0; i < array2.Length; i++)
{
    int number = int.Parse(array2[i]);
    if (set.Contains(number)) intersect.Add(number);
}

intersect.Sort();
for (int i = 0; i < intersect.Count; i++) Console.Write(intersect[i] + " ");